cpp/include/cudf/detail/null_mask.cuh

```diff
@@ -109,6 +112,116 @@ CUDF_KERNEL void offset_bitmask_binop(Binop op,
   if (threadIdx.x == 0) { atomicAdd(count_ptr, block_count); }
 }
 
+/**
+ * @brief Performs a segmented binary operation on bitmasks with configurable bit offsets
+ *
+ * This kernel applies a binary operation across multiple segments of bitmasks. Each segment is
```

Comment:
> This kernel applies a binary operation across multiple segments of bitmasks

I'm confused. Does this kernel performs operation for each segment separately, or perform one operation on multiple segments? For example, doing bit_and on all elements from two segments?